src/mam4xx/mo_setsox.hpp

```diff
@@ -0,0 +1,1341 @@
+#ifndef MAM4XX_MO_SETSOX_HPP
+#define MAM4XX_MO_SETSOX_HPP
+
+#include <ekat/util/ekat_math_utils.hpp>
+
+#include <haero/atmosphere.hpp>
+#include <haero/math.hpp>
+
+#include <mam4xx/aero_config.hpp>
+#include <mam4xx/conversions.hpp>
+#include <mam4xx/mam4_types.hpp>
+#include <mam4xx/utils.hpp>
+#include <mam4xx/wv_sat_methods.hpp>
+
+namespace mam4 {
+
+namespace mo_setsox {
+
+// mo_setsox-specific configuration
+struct Config {
+
+  // number of vertical levels
+  int pver = mam4::nlev;
+  int modeptr_accum = int(ModeIndex::Accumulation);
+  /*
+  these are the entries/indices in the gas-phase species array
+  from: eam/src/chemistry/pp_linoz_mam4_resus_mom_soag/mo_sim_dat.F90#L23)
+  solsym(: 31) = (/ 'O3', 'H2O2', 'H2SO4', 'SO2', 'DMS',  'SOAG', 'so4_a1',
+                    'pom_a1', 'soa_a1', 'bc_a1',  'dst_a1', 'ncl_a1',
+                    'mom_a1', 'num_a1', 'so4_a2',  'soa_a2', 'ncl_a2',
+                    'mom_a2', 'm_a2', 'dst_a3',  'ncl_a3', 'so4_a3', 'bc_a3',
+                    'pom_a3', 'soa_a3', 'mom_a3', 'num_a3', 'pom_a4',
+                    'bc_a4', 'mom_a4', 'num_a4' /)
+  Fortran-indexed:
+  1: 'O3', 2: 'H2O2', 3: 'H2SO4', 4: 'SO2', 5: 'DMS', 6: 'SOAG', 7: 'so4_a1',
+  8: 'pom_a1', 9: 'soa_a1', 10: 'bc_a1', 11: 'dst_a1', 12: 'ncl_a1',
+  13: 'mom_a1', 14: 'num_a1', 15: 'so4_a2', 16: 'soa_a2', 17: 'ncl_a2',
+  18: 'mom_a2', 19: 'num_a2', 20: 'dst_a3', 21: 'ncl_a3', 22: 'so4_a3',
+  23: 'bc_a3', 24: 'pom_a3', 25: 'soa_a3', 26: 'mom_a3', 27: 'num_a3',
+  28: 'pom_a4', 29: 'bc_a4', 30: 'mom_a4', 31: 'num_a4'
+  c++ indexed:
+  0: 'O3', 1: 'H2O2', 2: 'H2SO4', 3: 'SO2', 4: 'DMS', 5: 'SOAG', 6: 'so4_a1',
+  7: 'pom_a1', 8: 'soa_a1', 9: 'bc_a1', 10: 'dst_a1', 11: 'ncl_a1',
+  12: 'mom_a1', 13: 'num_a1', 14: 'so4_a2', 15: 'soa_a2', 16: 'ncl_a2',
+  17: 'mom_a2', 18: 'num_a2', 19: 'dst_a3', 20: 'ncl_a3', 21: 'so4_a3',
+  22: 'bc_a3', 23: 'pom_a3', 24: 'soa_a3', 25: 'mom_a3', 26: 'num_a3',
+  27: 'pom_a4', 28: 'bc_a4', 29: 'mom_a4', 30: 'num_a4'
+  */
+  int id_so2 = 3;
```

Comment:
At some point, we need to create a single source of truth for all these indices. Of course, things are too messy to do any reorganization at the moment. :-)